static/app/views/starfish/views/appStartup/screenSummary/deviceClassBreakdownBarChart.tsx

```diff
@@ -9,40 +10,109 @@ import {
 import {formatVersionAndCenterTruncate} from 'sentry/views/starfish/utils/centerTruncate';
+import {STARFISH_CHART_INTERVAL_FIDELITY} from 'sentry/views/starfish/utils/constants';
+import {appendReleaseFilters} from 'sentry/views/starfish/utils/releaseComparison';
+import {YAxis, YAXIS_COLUMNS} from 'sentry/views/starfish/views/screens';
+import {useTableQuery} from 'sentry/views/starfish/views/screens/screensTable';
+import {transformDeviceClassEvents} from 'sentry/views/starfish/views/screens/utils';
 
+const YAXES = [YAxis.COLD_START, YAxis.WARM_START];
 const XAXIS_CATEGORIES = ['high', 'medium', 'low', 'Unknown'];
-const CHART_HEIGHT = 70;
 
 interface DeviceClassBreakdownBarChartProps {
-  isError: boolean;
-  isLoading: boolean;
-  title: string;
-  yAxis: string;
-  data?: Series[];
+  additionalFilters?: string[];
+  chartHeight?: number;
 }
 
 function DeviceClassBreakdownBarChart({
-  isLoading,
-  isError,
-  title,
-  data,
-  yAxis,
+  chartHeight,
+  additionalFilters,
 }: DeviceClassBreakdownBarChartProps) {
+  const pageFilter = usePageFilters();
+  const location = useLocation();
+  const {query: locationQuery} = location;
   const {
     primaryRelease,
     secondaryRelease,
     isLoading: isReleasesLoading,
   } = useReleaseSelection();
 
+  const startType = decodeScalar(location.query[SpanMetricsField.APP_START_TYPE]) ?? '';
+  const yAxis = YAXIS_COLUMNS[startType === 'cold' ? YAxis.COLD_START : YAxis.WARM_START];
+  const query = new MutableSearch([...(additionalFilters ?? [])]);
+
+  const searchQuery = decodeScalar(locationQuery.query, '');
+  if (searchQuery) {
+    query.addStringFilter(prepareQueryForLandingPage(searchQuery, false));
+  }
+
+  const queryString = appendReleaseFilters(query, primaryRelease, secondaryRelease);
+
+  const {
+    data: startupDataByDeviceClass,
+    isLoading,
+    isError,
+  } = useTableQuery({
+    eventView: EventView.fromNewQueryWithPageFilters(
+      {
+        name: '',
+        fields: [
+          startType === 'cold'
+            ? 'avg(measurements.app_start_cold)'
+            : 'avg(measurements.app_start_warm)',
+          'device.class',
+          'release',
+        ],
+        yAxis: YAXES.map(val => YAXIS_COLUMNS[val]),
+        query: queryString,
+        dataset: DiscoverDatasets.METRICS,
+        version: 2,
+        interval: getInterval(
+          pageFilter.selection.datetime,
+          STARFISH_CHART_INTERVAL_FIDELITY
+        ),
+      },
+      pageFilter.selection
+    ),
+    enabled: !isReleasesLoading,
+    referrer: 'api.starfish.mobile-startup-bar-chart',
+    initialData: {data: []},
+  });
+
+  const transformedData: {} = transformDeviceClassEvents({
+    data: startupDataByDeviceClass,
+    yAxes: YAXES,
+    primaryRelease,
+    secondaryRelease,
+  });
+
+  const data = Object.values(
+    transformedData[
+      YAXIS_COLUMNS[startType === 'cold' ? YAxis.COLD_START : YAxis.WARM_START]
+    ]
+  ) as Series[];
```

Comment:
hmm do we need to type `transformedData` as `{}`?
Feels like you'll get `data` to be `Series[]` without having to do a type assertion if you remove `{}` from `transformedData`

Reply:
Oh you're totally right. I don't remember how that type snuck in there but thanks for catching it!